Functionality added.

Kgamer77 opened the pull request:
Added the BoardSpace object class for future use.
Finished the drawing of the tic-tac-toe board.
Implemented draw_circle
Implemented draw_cross
Added mouse clicking detection to play_game for later use and testing.

Kgamer77 committed:
Functionality added.

Added the BoardSpace object class for future use.
Finished the drawing of the tic-tac-toe board.
Implemented draw_circle
Implemented draw_cross
Added mouse clicking detection to play_game for later use and testing.

UliBob-omb reviewed (changes requested):
Maybe make the grid space state class use 1, 0, and -1 for the space's state instead of strings. 1 would be for player, 0 for empty, and -1 for AI/other player. This should make further changes for logic concerning the state of the spaces easier to work with.

UliBob-omb committed:
Normalized Line Endings

UliBob-omb committed:
Fixed a few Logic Errors

-play_game() should only be called from outside the class, else there is risk of bugs and loop recursion
-pygame.display.flip() should be called after changes to the display buffer has been made, otherwise the changes will not show.

UliBob-omb reviewed (approved):
A few logic errors fixed and normalized the line endings (which may have caused the borked diff file)
